File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mpf = 0;
        int prev = prices[0];        
        for(int i=1; i<prices.length; i++){
            if(prices[i]<prev) prev = prices[i];
            else mpf = Math.max(mpf,prices[i]-prev);
        }
        return mpf;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mpf: int = 0
        pp: int = prices[0] # purchase price
        for i in prices:
            if i-pp>mpf:
                mpf = i-pp
            elif pp>i:
                pp=i
        return mpf             